visualizer/IdealGraphVisualizer/SourceRepository/src/main/java/org/graalvm/visualizer/source/impl/ui/LocationOpener.java

```diff
@@ -65,22 +66,24 @@ private void openOrView(boolean focus) {
         int line = location.getLine();
         EditorCookie cake = toOpen.getLookup().lookup(EditorCookie.class);
 
-        Line l;
-        try {
-            l = cake.getLineSet().getOriginal(line - 1);
-        } catch (IndexOutOfBoundsException ex) {
-            // expected, the source has changed
-            return;
-        }
+        Line l = findLine(cake, line);
         if (l == null) {
             cake.open();
             StatusDisplayer.getDefault().setStatusText(Bundle.ERR_LineNotFound());
             return;
         }
-        if (SwingUtilities.isEventDispatchThread()) {
+        Mutex.EVENT.readAccess(() -> {
             l.show(Line.ShowOpenType.REUSE, focus ? Line.ShowVisibilityType.FRONT : Line.ShowVisibilityType.FRONT);
-        } else {
-            SwingUtilities.invokeLater(() -> l.show(Line.ShowOpenType.REUSE, focus ? Line.ShowVisibilityType.FRONT : Line.ShowVisibilityType.FRONT));
+        });
+    }
+
+    private Line findLine(EditorCookie cake, int line) {
+        try {
+            return cake.getLineSet().getOriginal(line - 1);
```

Comment:
Possibly IGV should honor https://www.graalvm.org/graphio/javadoc/jdk/graal/compiler/graphio/GraphLocations.html#locationOffsetStart(java.lang.Object) when the line isn't positive.